powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/PowerAuthEncryptionProviderBase.java

```diff
@@ -311,4 +399,23 @@ private EciesEncryptionContext extractEciesEncryptionContext(HttpServletRequest
         }
     }
 
+    /**
+     * Derive associated data.
+     * @param eciesScope ECIES scope.
+     * @param version Crypto protocol version.
+     * @param applicationKey Application key.
+     * @param activationId Activation ID.
+     * @return Associated data as byte array.
+     */
+    private byte[] deriveAssociatedData(EciesScope eciesScope, String version, String applicationKey, String activationId) {
+        if ("3.2".equals(version)) {
+            if (eciesScope == EciesScope.ACTIVATION_SCOPE) {
+                return ByteUtils.concatStrings(version, applicationKey, activationId);
+            } else {
+                return ByteUtils.concatStrings(version, applicationKey);
+            }
+        } else {
+            return null;
+        }
+    }
```

Comment:
This was already moved to `powerauth-crypto`, right?

Reply:
Yes, I will migrate this method.